File: Data Structures & Algorithms/concatenation-of-array/submission-5.cpp
class Solution {
public:
    vector<int> getConcatenation(vector<int>& nums) {
        const int n = nums.size();
        vector<int> res;
        res.reserve(2 * n);
        for (const int& num : nums) {
            res.push_back(num);
        }
        for (const int& num : nums) {
            res.push_back(num);
        }
        return res;
    }
};

File: Data Structures & Algorithms/maximum-subarray/submission-5.py
class Solution:
    def maxSubArray(self, nums: List[int]) -> int:
        if len(nums) == 1:
            return nums[0]
        M = float("-inf")
        m = float("inf")
        v = 0
        for num in nums:
            v += num
            if v > M:
                M = v
            if v < m:
                m = v
        return M - m